File: sp/src/main/java/wut/sp/Models/Image.java
package wut.sp.Models;

import wut.sp.Interfaces.IElement;

public class Image implements IElement{
    private String name;

    public Image() {
    }

    public Image(String name) {
        this.name = name;
    }

    public String getName() {
        return this.name;
    }

    @Override
    public void print() {
        System.out.println("Image with name: " + this.name);
    }

    @Override
    public void add(IElement element) {
        System.out.println("Cannot add element to image");
    }

    @Override
    public void remove(IElement element) {
        System.out.println("Cannot remove element from image");
    }

    @Override
    public IElement getElement(int index) {
        System.out.println("Cannot get element from image");
        return null;
    }
}

File: spC++/src/Implement/Models/Image.cpp
#include "Headers/Models/Image.hpp"
#include <unistd.h>

Image::Image(std::string imgName) {
    this->imageName = imgName;
}

Image::Image(Image* image) {
    this->imageName = image->imageName;
}

IElement *Image::add(IElement *element) {
    throw std::runtime_error("Cannot add element to image");
}

IElement *Image::remove(IElement *element) {
    throw std::runtime_error("Cannot remove element from image");
}

IElement *Image::get(int index) {
    throw std::runtime_error("Cannot get element from image");
}

IElement *Image::clone() {
    return new Image(this);
}

void Image::accept(IVisitor* v) {
    v->visitImage(this);
}

std::string Image::getImageName() {
    return this->imageName;
}